### PR TITLE
Get Rush ready to publish

### PR DESCRIPTION
**Pull request by nickpape-msft**
* Make the parallelized command required & default to false
* Move schema files to /src/schemas
* Move custom command line examples to /etc/samples
* Deprecate the 'test' command

### CODE REVIEW

File: apps/rush-lib/.npmignore [hunk 1]
```diff
@@ -21,4 +21,5 @@
 ## Project specific definitions
 # -----------------------------
 
-# (Add your exceptions here)
\ No newline at end of file
+# (Add your exceptions here)
+!/etc/**
```
Comment (pgonzal) [inline apps/rush-lib/.npmignore]: restrict this to examples

**Comment by pgonzal** [comment]
> // This example illustrates how to define your own custom command line

Fix the file name

---
Refers to: apps/rush-lib/etc/examples/Sample_CustomCommandLine.json:1 in 852295b. [](commit_id = 852295be7ee068b11e6b712db0cc095b62f9cd38, deletion_comment = False)

File: apps/rush-lib/etc/examples/Sample_CustomCommandLine.json [hunk 2]
```diff
@@ -7,19 +7,21 @@
...
 
 {
+  "$schema": "https://dev.office.com/json-schemas/rush/command-line.schema.json",
   "customCommands": [
     {
       "name": "check-copyright",
       "summary": "Ensure that all files have a copyright header",
-      "documentation": "Ensures that all TS files have a copyright header"
+      "documentation": "Ensures that all TS files have a copyright header",
+      "parallelized": false
```
Comment (pgonzal) [inline apps/rush-lib/etc/examples/Sample_CustomCommandLine.json]: >"parallelized": false [](start = 6, length = 21)

I thought this was the default now

File: apps/rush-lib/etc/examples/Sample_CustomCommandLine.json [hunk 3]
```diff
@@ -35,7 +37,7 @@
     "--color": {
       "optionType": "enum",
       "description": "Enable or disable colors in the build log, defaults to 'true'",
-      "supportedCommands": [ "build", "rebuild" ],
+      "associatedCommands": [ "build", "rebuild" ],
```
Comment (pgonzal) [inline apps/rush-lib/etc/examples/Sample_CustomCommandLine.json]: >rebuild [](start = 40, length = 7)

We should have an example of how to associate a custom option with a custom command.

Add check-copyright here.

File: apps/rush-lib/etc/examples/Sample_CustomCommandLine.json [hunk 4]
```diff
@@ -52,14 +54,14 @@
     "--ship": {
       "optionType": "flag",
       "description": "Perform a production build, including minification and localization steps",
-      "supportedCommands": [ "build", "rebuild" ],
+      "associatedCommands": [ "build", "rebuild", "check-copyright" ],
       "shortName": "-s"
     },
 
     "--minimal": {
```
Comment (pgonzal) [inline apps/rush-lib/etc/examples/Sample_CustomCommandLine.json]: >minimal [](start = 7, length = 7)

You can probably delete this.  It's already in the legacy example, and it's not illustrating anything new here.

**Comment by pgonzal** [comment]
>           "name": "true",

This is a pretty artificial example.  Why wouldn't --color just be a flag?  If you want to change the default, make it "--no-color"

---
Refers to: apps/rush-lib/etc/examples/Sample_CustomCommandLine.json:44 in 852295b. [](commit_id = 852295be7ee068b11e6b712db0cc095b62f9cd38, deletion_comment = False)

**Comment by pgonzal** [comment]
>     const parallelism: number = (this._parallelized ? this._parallelismParameter.value : 1);

(If it's not parallelized, do we still accept the "-p" parameter on the command line?)

---
Refers to: apps/rush-lib/src/cli/actions/CustomRushAction.ts:74 in 852295b. [](commit_id = 852295be7ee068b11e6b712db0cc095b62f9cd38, deletion_comment = False)

File: apps/rush-lib/etc/examples/Sample_CustomCommandLine.json (hunk 2 above)
Comment (nickpape-msft) [reply apps/rush-lib/etc/examples/Sample_CustomCommandLine.json]: It is a required parameter

Comment (nickpape-msft) [reply apps/rush-lib/etc/examples/Sample_CustomCommandLine.json]: I'll change it to true in the example

File: apps/rush-lib/etc/examples/Sample_CustomCommandLine.json (hunk 3 above)
Comment (nickpape-msft) [reply apps/rush-lib/etc/examples/Sample_CustomCommandLine.json]: I already added it below

File: apps/rush-lib/.npmignore (hunk 1 above)
Comment (nickpape-msft) [reply apps/rush-lib/.npmignore]: done

---
In reply to: [148912766](https://github.com/Microsoft/web-build-tools/pull/431#discussion_r148912766) [](ancestors = 148912766)

File: apps/rush-lib/etc/examples/Sample_CustomCommandLine.json (hunk 2 above)
Comment (nickpape-msft) [reply apps/rush-lib/etc/examples/Sample_CustomCommandLine.json]: This is the required parameter. I will change it to true.

---
In reply to: [148912952](https://github.com/Microsoft/web-build-tools/pull/431#discussion_r148912952) [](ancestors = 148912952)

File: apps/rush-lib/etc/examples/Sample_CustomCommandLine.json (hunk 3 above)
Comment (nickpape-msft) [reply apps/rush-lib/etc/examples/Sample_CustomCommandLine.json]: It is already associated with --ship

---
In reply to: [148913014](https://github.com/Microsoft/web-build-tools/pull/431#discussion_r148913014) [](ancestors = 148913014)

**Comment by nickpape-msft** [comment]
>     const parallelism: number = (this._parallelized ? this._parallelismParameter.value : 1);

I'll fix it now

---
In reply to: [341852514](https://github.com/Microsoft/web-build-tools/pull/431#issuecomment-341852514) [](ancestors = 341852514)

---
Refers to: apps/rush-lib/src/cli/actions/CustomRushAction.ts:74 in 852295b. [](commit_id = 852295be7ee068b11e6b712db0cc095b62f9cd38, deletion_comment = False)

**Comment by pgonzal** [comment]
:shipit:

**Comment by iclanton** [comment]
Approved.

**Pull request merged**